crates/generate_artifacts/src/operation_text.rs

```diff
@@ -0,0 +1,54 @@
+use common_lang_types::{DocumentId, IsographObjectTypeName, QueryOperationName, QueryText};
+use intern::string_key::Intern;
+use isograph_config::{PersistedDocumentsHashAlgorithm, PersistedDocumentsOptions};
+use md5::{Digest, Md5};
+use sha2::Sha256;
+
+use crate::generate_artifacts::OperationText;
+
+pub(crate) fn generate_operation_text(
+    query_name: QueryOperationName,
+    operation_name: IsographObjectTypeName,
+    query_text: QueryText,
+    persisted_documents_options: Option<&PersistedDocumentsOptions>,
+    indentation_level: u8,
+) -> (OperationText, Vec<(DocumentId, QueryText)>) {
+    let indent = "  ".repeat((indentation_level + 1) as usize);
+    let mut persisted_documents = vec![];
```

Comment:
In the non-persisted case, we create an empty vec and pass that around, and then drop it. That's no big deal.

But it does open us up to bugs where the developer might mess up that logic and generate a persisted queries json file without all of the expected persisted queries present.

So let's restructure this to accept an `Option<(&PersistedDocumentsOptions, &mut Vec<PersistedDocumentEntry>)>`, so that it's harder to make that mistake